kyaml/fn/runtime/runtimeutil/functiontypes.go

```diff
@@ -33,6 +34,78 @@ const (
 	UserNobody ContainerUser = "nobody"
 )
 
+const defaultEnvValue string = "true"
+
+// ContainerEnvs contains the environment variables for the container
+type ContainerEnvs struct {
+	// EnvsMap is a key-value map that will be set as env in container
+	EnvsMap map[string]string `json:"envsMap,omitempty" yaml:"envsMap,omitempty"`
+
+	// ExportKeys are only env key. Value will be the value in the host system
+	ExportKeys []string `json:"exportKeys,omitempty" yaml:"exportKeys,omitempty"`
+}
+
+// GetDockerFlags returns docker run style env flags
+func (ce *ContainerEnvs) GetDockerFlags() []string {
+	envs := ce.EnvsMap
+	if envs == nil {
+		envs = make(map[string]string)
+	}
+
+	// default envs
+	envs["LOG_TO_STDERR"] = defaultEnvValue
+	envs["STRUCTURED_RESULTS"] = defaultEnvValue
```

Comment:
Line 50 is a map ptr copy, not a map copy, so these lines mutate the instance.  

A method named `GetWhatever` should not be a mutator.

To set up defaults, define a factory function like

 func MakeContainerEnv() *ContainerEnv {
   // install defaults.
}

Reply:
There is a factory function in line 103. Will add the default values there.

Reply:
I am still not very sure about which is better in a factory method, return a pointer or a value?

Reply:
I don't think we've been strict here.

I want to say return values for "small", immutable structs (like this one seems to be) and pointers for mutable structs or for structs so big that coping would be a problem.  

But on the whole, kustomize typically passes around pointers.